### PR TITLE
NIFI-4862 Add Copy original attributtes to SelectHiveQL processer

### PR DESCRIPTION
**Pull request by jles01**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java [hunk 1]
```diff
@@ -306,6 +319,7 @@ private void onTrigger(final ProcessContext context, final ProcessSession sessio
         final boolean quote = context.getProperty(HIVEQL_CSV_QUOTE).asBoolean();
         final boolean escape = context.getProperty(HIVEQL_CSV_HEADER).asBoolean();
         final String fragmentIdentifier = UUID.randomUUID().toString();
+        final boolean copyAttributtes = context.getProperty(HIVEQL_COPY_ATRIBUTTES).asBoolean();
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: Please correct typo in the variable name -- `copyAttributes`.

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java [hunk 2]
```diff
@@ -437,6 +451,16 @@ private void onTrigger(final ProcessContext context, final ProcessSession sessio
                         resultSetFlowFiles.set(i,
                                 session.putAttribute(resultSetFlowFiles.get(i), "fragment.count", Integer.toString(fragmentIndex)));
                     }
+
+                    if(copyAttributtes)
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: Please follow the [style guide located here](https://cwiki.apache.org/confluence/display/NIFI/Contributor+Guide#ContributorGuide-CodeStyle) for consistency.

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java [hunk 3]
```diff
@@ -437,6 +451,16 @@ private void onTrigger(final ProcessContext context, final ProcessSession sessio
...
+                    {
+                        for(Map.Entry<String, String> entry : fileToProcess.getAttributes().entrySet()) {
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: The attributes on the original flowfile are static, but this logic is placed inside a loop which iterates over every resulting flowfile (there may be many) and performs the same extraction multiple times. Please move the retrieval of the original attributes outside of the loop and store them in a local variable, then use `session.putAllAttributes(resultSetFlowFiles.get(i), map);` inside the loop to copy all the original attributes into the new flowfile.

**Comment by alopresto** [comment]
Thanks for your first contribution to Apache NiFi. I have made some inline comments on the code. Please also add a unit test covering the new functionality you added. You can see existing unit tests for this code in [`TestSelectHiveQL`](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/test/java/org/apache/nifi/processors/hive/TestSelectHiveQL.java#L63). Our [Contributor Guide](https://cwiki.apache.org/confluence/display/NIFI/Contributor+Guide) also has helpful information to make your submission review go as smoothly as possible.

**Comment by jles01** [comment]
Hi,

I follwed your guidelines and corrected the problems which you highlighted.

Best regards,
Jakub

**Comment by mattyb149** [comment]
It doesn't look like this branch was successfully rebased against the latest master, can you try to rebase such that only your new commits show up in the PR? If all else fails, you can create a separate branch based off this current one, then reset this branch to the latest master, then cherry-pick the new commits from the copied branch, then force push this branch up (so the PR gets updated).

**Comment by ijokarumawak** [comment]
Hi @jles01 , as I wrote on the Apache JIRA [NIFI-4862](https://issues.apache.org/jira/browse/NIFI-4862), Matt's PR has been merged in order to achieve the same goal with yours. Would you check #2605 to see if there's anything to be added? If that one looks good to your needs, please close this PR. Thanks for your contribution!

**Pull request closed without merging**